client/js/components/StatsLeaderboard.vue

```diff
@@ -0,0 +1,359 @@
+<template>
+  <div>
+    <div id="stats-table-upper-surface" class="d-flex align-end">
+      <v-select
+        v-model="selectedMetric"
+        :items="metricChoices"
+        label="Select Metric"
+        class="filter-select mr-4 fit"
+        data-cy="metric-select"
+      />
+      <v-select
+        v-model="selectedWeeks"
+        :items="weeks"
+        label="Select Weeks"
+        data-cy="week-select"
+        class="flex-grow-1"
+        menu-props="data-week-select-menu"
+        multiple
+        chips
+        deletable-chips
+      />
+    </div>
+    <v-data-table
+      :items="tableRows"
+      :headers="tableColumns"
+      :sort-by.sync="sortBy"
+      :loading="loading"
+      :item-class="tableRowClass"
+    >
+      <template #[`item.rank`]="{item, value}">
+        <span :data-rank="item.username">{{ value }} </span>
+      </template>
```

Comment:
Checkout the docs for the `item.name` slot in data tables: https://vuetifyjs.com/en/api/v-data-table/#slots-item.%3Cname%3E

This customizes the cells in the `rank` column just so I can apply the `data-rank` attribute for selection during testing